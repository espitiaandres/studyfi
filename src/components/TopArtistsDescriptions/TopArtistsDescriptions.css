.tooltips {
    max-width: 400px;
}

.tooltip {
    background: #1ED760;
    border: none;
    border-radius: 4px;
    display: inline-block;
    font-size: 16px;
    margin: 0px 4px;  
    padding: 2px 10px 4px 10px;  
    text-decoration: none;
}

.tooltip:hover {
    animation: fadeIn ease 0.7s;
    -webkit-animation: fadeIn ease 0.7s;
    -moz-animation: fadeIn ease 0.7s;
    -o-animation: fadeIn ease 0.7s;
    -ms-animation: fadeIn ease 0.7s;
    background: black;
    color: white;
}

.seasonStyling {
    background: #651B0E;
}

.seasonStylingAlt {
    background: #E25B6A;
}
