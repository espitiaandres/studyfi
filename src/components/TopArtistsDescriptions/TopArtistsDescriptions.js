//
//  TopArtistsDescriptions.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import ReactTooltip from "react-tooltip";
import './TopArtistsDescriptions.css';
import { 
    followersDesc,
    genreDesc,
    nameDesc,
    photoDesc,
    popularityDesc
} from '../../utils/tableFormat';

const TopArtistsDescriptions = ({ season }) => {
    let seasonStyling = season ? "seasonStyling" : "";
    let seasonStylingAlt = season ? "seasonStylingAlt" : "";

    return (
        <div className="columnDescriptions">
            <ReactTooltip className="tooltips" id="name" type="light" effect="solid" place="top">
                <span>{nameDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="name">
                name
            </button>

            <ReactTooltip className="tooltips" id="followers" type="light" effect="solid" place="top">
                <span>{followersDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStylingAlt}`} data-tip data-for="followers">
                followers
            </button>

            <ReactTooltip className="tooltips" id="genre" type="light" effect="solid" place="top">
                <span>{genreDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="genre">
                genres
            </button>

            <ReactTooltip className="tooltips" id="photo" type="light" effect="solid" place="top">
                <span>{photoDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStyling}`} data-tip data-for="photo">
                photo
            </button>

            <ReactTooltip className="tooltips" id="popularity" type="light" effect="solid" place="top">
                <span>{popularityDesc}</span>
            </ReactTooltip>
            <button className={`tooltip ${seasonStylingAlt}`} data-tip data-for="popularity">
                popularity
            </button>
        </div>
    )
}

export default TopArtistsDescriptions;
