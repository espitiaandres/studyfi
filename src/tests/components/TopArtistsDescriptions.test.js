//
//  TopArtistsDescriptions.test.js
//  react-spotify-player
//
//  Created by Andres Espitia.
//  Copyright Â© 2020 Andres Espitia. All rights reserved.
//

import React from 'react';
import { shallow } from 'enzyme';
import TopArtistsDescriptions from '../../components/TopArtistsDescriptions/TopArtistsDescriptions';

const topArtistsDescriptionsPropsTest = {
    season: true
}

test('should render TopArtistsDescriptions component correctly', () => {
    const wrapper = shallow(<TopArtistsDescriptions />);
    expect(wrapper).toMatchSnapshot();
});

test('should render TopArtistsDescriptions component correctly', () => {
    const wrapper = shallow(
        <TopArtistsDescriptions 
            season={topArtistsDescriptionsPropsTest.season}
        />
    );
    expect(wrapper).toMatchSnapshot();
});
