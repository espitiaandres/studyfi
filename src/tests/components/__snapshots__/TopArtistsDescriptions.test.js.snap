// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render TopArtistsDescriptions component correctly 1`] = `ShallowWrapper {}`;

exports[`should render TopArtistsDescriptions component correctly 2`] = `ShallowWrapper {}`;
